lib/matplotlib/cbook.py

```diff
@@ -2244,6 +2244,21 @@ def _reshape_2D(X):
     return X
 
 
+def ensure_3d(arr):
```

Comment:
https://www.python.org/dev/peps/pep-0008/#id29

It means that it is to be treated as being for internal use only. This
gives developers the freedom to modify the API of such modules or functions
without warning or deprecation cycles. As noted later in PEP8, it is easier
to start with something as private and then make it public later, as
opposed to releasing it as public and then modifying it later.

On Mon, Sep 28, 2015 at 10:23 AM, Jouni K. Seppänen <
notifications@github.com> wrote:

> In lib/matplotlib/cbook.py
> <https://github.com/matplotlib/matplotlib/pull/5106#discussion_r40557282>:
>
> > @@ -2244,6 +2244,21 @@ def _reshape_2D(X):
> >      return X
> >
> >
> > +def ensure_3d(arr):
>
> I would have thought that an underscore in the beginning means a function
> that is only called from within cbook, but clearly some underscored
> functions are called from elsewhere. What is the intended meaning?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5106/files#r40557282>.
>